experimental/iterators/lib/Dialects/Tabular/IR/Tabular.cpp

```diff
@@ -0,0 +1,124 @@
+
+void TabularDialect::initialize() {
+#define GET_OP_LIST
+  addOperations<
+#include "iterators/Dialect/Tabular/IR/TabularOps.cpp.inc"
+      >();
+  addTypes<
+#define GET_TYPEDEF_LIST
+#include "iterators/Dialect/Tabular/IR/TabularOpsTypes.cpp.inc"
+      >();
+}
+
+//===----------------------------------------------------------------------===//
+// Tabular interfaces
+//===----------------------------------------------------------------------===//
+
+#include "iterators/Dialect/Tabular/IR/TabularOpInterfaces.cpp.inc"
+#include "iterators/Dialect/Tabular/IR/TabularTypeInterfaces.cpp.inc"
+
+//===----------------------------------------------------------------------===//
+// Tabular operations
+//===----------------------------------------------------------------------===//
+
+#define GET_OP_CLASSES
+#include "iterators/Dialect/Tabular/IR/TabularOps.cpp.inc"
+
+LogicalResult CreateTabularViewOp::verify() {
+  auto viewType = view().getType().cast<TabularViewType>();
+  TypeRange columnTypes = viewType.getColumnTypes();
+
+  // Verify matching column/element types.
+  if (!llvm::all_of_zip(memrefs().getTypes(), columnTypes,
+                        [](Type t1, Type t2) {
+                          return t1.cast<MemRefType>().getElementType() == t2;
+                        })) {
+
+    std::string viewColumnTypes;
+    {
+      llvm::raw_string_ostream stream(viewColumnTypes);
+      llvm::interleaveComma(columnTypes, stream,
+                            [&](Type type) { type.print(stream); });
+    }
+    std::string memrefColumnTypes;
+    {
+      llvm::raw_string_ostream stream(memrefColumnTypes);
+      llvm::interleaveComma(memrefs().getTypes(), stream, [&](Type type) {
+        type.cast<MemRefType>().getElementType().print(stream);
+      });
+    }
+    return emitOpError()
+           << "type mismatch: should return a tabular view with the column "
+           << "types of the elements of the input memrefs (namely: "
+           << "tabular_view<" << memrefColumnTypes << ">) "
+           << "but returns a tabular_view<" << viewColumnTypes << ">.";
+  }
+
+  // Verify memrefs have rank 1 and contiguous memory layout.
+  {
+    for (const auto &indexedMemref : llvm::enumerate(memrefs().getTypes())) {
+      // Must have static rank = 1.
+      auto memrefType = indexedMemref.value().cast<MemRefType>();
+      if (memrefType.getRank() != 1)
+        return emitOpError()
+               << "unsupported type: input memref #" << indexedMemref.index()
+               << " has rank != 1, which is not supported.";
+
+      // Only support contiguous memory layout.
+      if (!isStaticShapeAndContiguousRowMajor(memrefType))
+        return emitOpError()
+               << "unsupported type: input memref #" << indexedMemref.index()
+               << " does not have a static shape and a contiguous memory "
+                  "layout, both of which are required.";
+    }
+
+    // Verify all memrefs are of equal static length.
```

Comment:
please check OpBase.td for support for similar things (not sure it exists but at least let's check)